$OpenBSD: patch-src_slowhttptest_cc,v 1.3 2019/11/08 15:59:28 fcambus Exp $

Check that the argument passed to freeaddrinfo(3) is not NULL.

Index: src/slowhttptest.cc
--- src/slowhttptest.cc.orig
+++ src/slowhttptest.cc
@@ -152,7 +152,9 @@ SlowHTTPTest::SlowHTTPTest(int delay, int duration, 
 }
 
 SlowHTTPTest::~SlowHTTPTest() {
-  freeaddrinfo(addr_);
+  if (addr_) {
+    freeaddrinfo(addr_);
+  }
 
   for(std::vector<StatsDumper*>::iterator i = dumpers_.begin();
        i != dumpers_.end(); ++i) {
